File: Algo-DS-Python/2. Recursion/sumOfDigit.py
'''
que: How to find the sum of digits of a positive 
    integer number using recursion?
'''

def sumOfDigit(n):
    assert n>=0 and int(n) == n, 'number must be positive integer'
    if n == 0:
        return 0
    else:
        return (n%10) + sumOfDigit(n//10)

print(sumOfDigit(223))   # 2 + 2 + 3 = 7

File: DSA-JAVA/1_recursion/sumOfDigit.java
public class sumOfDigit {
    public static void main(String[] args) {
        int x = sum(5113);
        System.out.println(x);  // 5+1+1+3 = 10
    }   
    
    public static int sum(int n) {
        if(n == 0) return 0;
        return (n%10) + sum(n/10);
    }
}
